Refactor datasync

ondrej-fabry opened the pull request:
Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

milanlenco commented on datasync/syncbase/msg/change_event.go:
```diff
@@ -23,14 +23,37 @@ import (
+	}
+}
+
+type ChangeEvent struct {
```
Transaction options will be added here?

ondrej-fabry committed:
Make change events returns list of changes to support transactions

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

Commit:
Merge pull request #326 from milanlenco/kvscheduler

KVscheduler

ondrej-fabry committed:
Merge API definitions for resync

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

ondrej-fabry committed:
Cleanup resync package

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

Commit:
Merge pull request #341 from milanlenco/kvscheduler

Kvscheduler: support circular deps (veths), fix txn reverting and minor bug fixes

ondrej-fabry committed:
General cleanup for datasync & db packages

- remove unnecessary interface `LazyValueWithRev` from datasync API
  and use its super type `KeyVal`
- un-export types that are used only internally
- change default timeouts from constants to variables and export them
- correctly return or log some errors that were ignored before
- remove unused types and fields
- merge few smaller files together
- improve wording and log levels in several logs

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

ondrej-fabry committed:
Satisfy linter

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

ondrej-fabry committed:
Merge remote-tracking branch 'upstream/dev' into refactor-datasync

ondrej-fabry committed:
Merge remote-tracking branch 'milan/kvscheduler' into refactor-datasync

ondrej-fabry committed:
Merge pull request #343 from milanlenco/kvscheduler

Kvscheduler revisited

ondrej-fabry committed:
Merge remote-tracking branch 'upstream/dev' into refactor-datasync

ondrej-fabry committed:
Remove scheduler workaround for datasync and use watcher

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

ondrej-fabry committed:
Satisfy linter

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>

ondrej-fabry committed:
Remove kvscheduler from cn-infra

Signed-off-by: Ondrej Fabry <ofabry@cisco.com>
